### PR TITLE
Remove dangling TODO that links to a resolved issue

### PR DESCRIPTION
**Pull request by grantatspothero**
The upstream issue: https://github.com/trinodb/trino/issues/9021 cannot be reproduced, see the issue for details

### CODE REVIEW

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java [hunk 1]
```diff
@@ -311,9 +311,6 @@ private JdbcTableHandle flushAttributesAsQuery(ConnectorSession session, JdbcTab
 
         List<JdbcColumnHandle> newColumnsList = newColumns.build();
 
-        // TODO(https://github.com/trinodb/trino/issues/9021) We are reading all grouping columns from remote database as at this point we are not able to tell if they are needed up in the query.
-        // As a reason of that we need to also have matching column handles in JdbcTableHandle constructed below, as columns read via JDBC must match column handles list.
-        // For more context see assertion in JdbcRecordSetProvider.getRecordSet
```
Comment (kasiafi) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java:316]: I think that we should only remove the TODO as irrelevant: `TODO(https://github.com/trinodb/trino/issues/9021) `.
The remaining path of the comment explains a change, which looks like an inconsistency fix. @alexjo2144  is this right?

**Review by wendigo — APPROVED** [review]

**Comment by grantatspothero** [comment]
Forgot about this tiny PR sorry, applied feedback.

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java (hunk 1 above)
Comment (grantatspothero) [reply plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java]: Sounds good 👍

**Review by hashhar — APPROVED** [review]
Thanks.

**Pull request merged**